### PR TITLE
Python 3: port tests in resource-timing

### PR DESCRIPTION
**Pull request by ziransun**

### CODE REVIEW

**Comment by ziransun** [comment]
@stephenmcgruer @Hexcles  any chance to have a look first?

File: resource-timing/SyntheticResponse.py [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-        elif arg.startswith("mime:"):
-            headers.append(("Content-Type", urllib.unquote(arg[5:])))
-        elif arg.startswith("send:"):
-            text = urllib.unquote(arg[5:])
+        elif arg.startswith(u"redirect:"):
+            return (302, u"WEBPERF MARKETING"), [(b"Location", unquote(arg[9:]).encode("iso-8859-1"))], u"TEST"
+        elif arg.startswith(u"mime:"):
+            headers.append((b"Content-Type", unquote(arg[5:]).encode("iso-8859-1")))
+        elif arg.startswith(u"send:"):
+            text = unquote(arg[5:]).encode("iso-8859-1")
```
[the remark this answers is not in the document]
Comment (ziransun) [reply resource-timing/SyntheticResponse.py:20]: Yes, encode here is not necessary.

File: resource-timing/resources/cors-ahem.py [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-    response.headers.set("Timing-Allow-Origin", "*")
-    response.headers.set("ETag", etag)
-    font = "../../fonts/Ahem.ttf"
+    response.headers.set(b"Cache-Control", b"public, max-age=86400")
+    response.headers.set(b"Content-Type", b"font/truetype")
+    response.headers.set(b"Access-Control-Allow-Origin", b"*")
+    response.headers.set(b"Timing-Allow-Origin", b"*")
+    response.headers.set(b"ETag", etag)
+    font = u"../../fonts/Ahem.ttf"
     path = os.path.join(os.path.dirname(__file__), font)
```
[the remark this answers is not in the document]
Comment (Hexcles) [reply resource-timing/resources/cors-ahem.py:16]: `__file__` is a native string. `os.path` works with both strings as long as they are not mixed, so here I think we should call `isomorphic_decode(__file__)` to get a text string, consistent with `font`.

**Comment by ziransun** [comment]
Thanks @stephenmcgruer!

File: resource-timing/SyntheticResponse.py [hunk 3]
```diff
@@ -1,23 +1,23 @@
-import urllib
+from six.moves.urllib.parse import unquote
 import sleep
 
 def main(request, response):
-    index = request.request_path.index("?")
-    args = request.request_path[index+1:].split("&")
+    index = request.request_path.index(u"?")
```
[the remark this answers is not in the document]
Comment (ziransun) [reply resource-timing/SyntheticResponse.py]: @Hexcles: so for native string I take it's text in Python 3, bytes in Python 2? Please let me know if the updated code here makes sense.

[the remark this answers is not in the document]
Comment (Hexcles) [reply resource-timing/SyntheticResponse.py]: That's right. I agree we need to call `isomorphic_encode` here for now.

File: resource-timing/SyntheticResponse.py [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
+        elif arg.endswith(u"ms"):
             sleep.sleep_at_least(float(arg[0:-2]))
-        elif arg.startswith("redirect:"):
-            return (302, "WEBPERF MARKETING"), [("Location", urllib.unquote(arg[9:]))], "TEST"
-        elif arg.startswith("mime:"):
-            headers.append(("Content-Type", urllib.unquote(arg[5:])))
-        elif arg.startswith("send:"):
-            text = urllib.unquote(arg[5:])
+        elif arg.startswith(u"redirect:"):
+            return (302, u"WEBPERF MARKETING"), [(b"Location", unquote(arg[9:]).encode("iso-8859-1"))], u"TEST"
```
[the remark this answers is not in the document]
Comment (stephenmcgruer) [reply resource-timing/SyntheticResponse.py]: (Unresolving just to track that this question exists; the code has been updated to use `isomorphic_decode` for now)

[the remark this answers is not in the document]
Comment (ziransun) [reply resource-timing/SyntheticResponse.py]: Can we mark this as "resolved" now?

[the remark this answers is not in the document]
Comment (stephenmcgruer) [reply resource-timing/SyntheticResponse.py]: I just marked it as unresolved 10 minutes ago ;)

To be clear - the *code* seems fine to me, but @Hexcles had a question for @yoavweiss that has not been answered and I want to track that that exists. This doesn't preclude this PR from landing.

[the remark this answers is not in the document]
Comment (Hexcles) [reply resource-timing/SyntheticResponse.py]: Oh thanks for catching this, Stephen. If this is the only remaining open comment, let's land this PR as is and open an issue to track.

**Review by stephenmcgruer — APPROVED** [review]

**Comment by stephenmcgruer** [comment]
Taskcluster failure is stability check in Firefox for navigation-timing/nav2_test_redirect_server.html failing, looking at wpt.fyi historical results its already flaky.

**Pull request merged**